### PR TITLE
Update detection notebook

### PR DESCRIPTION
**Pull request by QRemy**
Extend the notebook to show more of gammapy.detect,  detail how to :
- perform adaptive smoothing on counts image
- produce 2-dimensional test-statistics (TS)
- run a peak finder to detect point-source candidates
- compute Li & Ma singificance images
- estimate source candidates radius and excess counts

Address issues #2593 and #752

### CODE REVIEW

File: tutorials/detect.ipynb [hunk 1]
```diff
@@ -8,22 +8,25 @@
...
+    "We will use images from a Fermi-LAT 3FHL high-energy Galactic center dataset to do this:\n",
     "\n",
-    "* produce 2-dimensional test-statistics (TS) images using Fermi-LAT 3FHL high-energy Galactic center dataset\n",
-    "* run a peak finder to make a source catalog\n",
-    "* do some simple measurements on each source\n",
-    "* compare to the 3FHL catalog\n",
+    "* perform adaptive smoothing on counts image\n",
+    "* produce 2-dimensional test-statistics (TS)\n",
+    "* run a peak finder to detect point-source candidates\n",
+    "* compute Li & Ma singificance images\n",
```
Comment (registerrier) [inline tutorials/detect.ipynb]: significance

File: tutorials/detect.ipynb [hunk 2]
```diff
@@ -42,7 +45,21 @@
    "outputs": [],
    "source": [
     "%matplotlib inline\n",
-    "import matplotlib.pyplot as plt"
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib.cm as cm\n",
+    "\n",
+    "# defalut matplotlib colors without grey\n",
+    "colors = [\n",
```
Comment (registerrier) [inline tutorials/detect.ipynb]: this might be a bit confusing here no?

File: tutorials/detect.ipynb [hunk 3]
```diff
@@ -87,6 +114,40 @@
     ")"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Adaptive smoothing\n",
```
Comment (registerrier) [inline tutorials/detect.ipynb]: Maybe this specific question deserves a small notebook on its own. Something like how to produce an adaptively smoothed map? 
We probably need to re-discuss how to present how-to's that do not explicitly belong to a specific tutorial.

File: tutorials/detect.ipynb [hunk 4]
```diff
@@ -139,9 +200,10 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Source catalog\n",
+    "## Source candidates\n",
     "\n",
-    "Let's run a peak finder on the `sqrt_ts` image to get a list of sources (positions and peak `sqrt_ts` values)."
+    "Let's run a peak finder on the `sqrt_ts` image to get a list of point-sources candidates (positions and peak `sqrt_ts` values).\n",
+    "The `find_peaks` function perform a local maximun search in a sliding window, the argument `min_distance` is the minimum pixel distance between peaks (smallest possible value and default is 1 pixel)."
```
Comment (registerrier) [inline tutorials/detect.ipynb]: performs a local...

File: tutorials/detect.ipynb [hunk 5]
```diff
@@ -87,6 +114,40 @@
...
+    "images[\"counts\"].plot(add_cbar=True, vmax=10);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## TS map estimation\n",
+    "\n",
+    "The TS map is computed fitting by a single parameter amplitude fit as described in Appendix A of [Stewart (2009)](https://ui.adsabs.harvard.edu/abs/2009A%26A...495..989S/abstract). The fit is simplified by finding roots of the derivative of the fit statistics (default settings use [Brent's method](https://en.wikipedia.org/wiki/Brent%27s_method))."
```
Comment (registerrier) [inline tutorials/detect.ipynb]: You might expand a bit here to have a more pedagogical tutorial.

File: tutorials/detect.ipynb [hunk 6]
```diff
@@ -181,9 +244,10 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Measurements\n",
+    "## Li & Ma significance maps\n",
```
Comment (registerrier) [inline tutorials/detect.ipynb]: Maybe here, the interesting point is rather to go beyond the point source assumption that is made by the `TSMapEstimator`

File: tutorials/detect.ipynb [hunk 7]
```diff
@@ -181,9 +244,10 @@
...
     "\n",
-    "* TODO: show cutout for a few sources and some aperture photometry measurements (e.g. energy distribution, significance, flux)"
+    "Additionaly we can compute significance for an observed number of counts and known background using equation (17) from the [Li & Ma (1983)](https://ui.adsabs.harvard.edu/abs/1983ApJ...272..317L/abstract). We can perform this calculation intergating the counts within different radius. To do so we use an astropy Tophat kernel with the `compute_lima_image` function.\n",
```
Comment (registerrier) [inline tutorials/detect.ipynb]: Here you should say that this is an extension of the Li & Ma formula for known background.

File: tutorials/detect.ipynb (hunk 6 above)
Comment (QRemy) [reply tutorials/detect.ipynb]: I added a transition comment at the end of the TS map part about extended sources:

> Note that we used the instrument point-spread-function (PSF) as kernel, so the hypothesis we test is the presence of a point source. In order to test for extended sources we would have to use as kernel an extended template convolved by the PSF. Alternatively, we can compute the significance of an extended excess using the Li & Ma formalism, which is faster as no fitting is involve.

File: tutorials/detect.ipynb (hunk 5 above)
Comment (QRemy) [reply tutorials/detect.ipynb]: Added TS definition and reference

File: tutorials/detect.ipynb (hunk 6 above)
Comment (registerrier) [reply tutorials/detect.ipynb]: Perfect

**Pull request merged**